src/storm/mesos/MesosNimbus.java

```diff
@@ -97,57 +105,93 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
-      String framework_name = Optional.fromNullable((String) conf.get(CONF_MESOS_FRAMEWORK_NAME)).or("Storm!!!");
-
-      FrameworkInfo.Builder finfo = FrameworkInfo.newBuilder()
-          .setName(framework_name)
-          .setFailoverTimeout(failoverTimeout.doubleValue())
-          .setUser("")
-          .setRole(role)
-          .setCheckpoint(checkpoint);
-
-      if (id != null) {
-        finfo.setId(FrameworkID.newBuilder().setValue(id).build());
-      }
-
-      Integer port = (Integer) _conf.get(CONF_MESOS_LOCAL_FILE_SERVER_PORT);
-      LOG.info("Using local port: " + port);
-      _localFileServerPort = Optional.fromNullable(port);
-
-      _httpServer = new LocalFileServer();
-      _configUrl = _httpServer.serveDir("/conf", "conf", _localFileServerPort);
-      LOG.info("Started serving config dir under " + _configUrl);
-
-      MesosSchedulerDriver driver =
-          new MesosSchedulerDriver(
-              _scheduler,
-              finfo.build(),
-              (String) conf.get(CONF_MASTER_URL));
-
-      driver.start();
-      LOG.info("Waiting for scheduler to initialize...");
-      _scheduler.waitUntilRegistered();
-      LOG.info("Scheduler initialized...");
+        initialize(conf,localDir);
+
+        MesosSchedulerDriver driver = createMesosDriver();
+        
+        driver.start();
+      
+        LOG.info("Waiting for scheduler to initialize...");
+        
+        getScheduler().waitUntilRegistered();
+        
+        LOG.info("Scheduler initialized...");
+      
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
-
+  
+  @SuppressWarnings("unchecked")
+  protected void initialize(Map conf, String localDir) throws Exception {
+      
+      setConfig(conf);
+      setLocalState(new LocalState(localDir));
+
+      _allowedHosts = listIntoSet((List<String>)conf.get(CONF_MESOS_ALLOWED_HOSTS));
+      _disallowedHosts = listIntoSet((List<String>)conf.get(CONF_MESOS_DISALLOWED_HOSTS));
+      
+      setScheduler(new NimbusScheduler());
+      createLocalServerPort();
+      setupHttpServer();
+  } 
+  protected void setLocalState(LocalState state) {
+      _state = state;
+  }
+  protected LocalState getLocalState() {
+      return _state;
+  }
+  protected void createLocalServerPort() {
+    Integer port = (Integer) _conf.get(CONF_MESOS_LOCAL_FILE_SERVER_PORT);
+    LOG.info("Using local port: " + port);
+    setLocalServerPort(Optional.fromNullable(port)); 
+  }
+  protected void setLocalServerPort(Optional<Integer> port) {
+      _localFileServerPort = port;
+  }
+  protected Optional<Integer> getLocalServerPort(){
+      return _localFileServerPort;
+  }
+  protected void setScheduler(NimbusScheduler scheduler) {
+      _scheduler = scheduler;
+  }
+  protected NimbusScheduler getScheduler() {
+      return _scheduler;
+  }
+  protected void setupHttpServer() throws Exception {
```

Comment:
@pdread100 I think your refactoring is reasonable, but then it makes this commit a lot harder to merge now since there are a lot of changes and also some changes are related to a new feature and mostly are just refactoring.